app/src/main/java/org/greenstand/android/TreeTracker/capture/TreeCaptureViewModel.kt

```diff
@@ -26,7 +27,7 @@ class TreeCaptureViewModel(
 
     suspend fun captureLocation() {
         _state.value = _state.value?.copy(isGettingLocation = true)
-        treeCapturer.pinLocation()
+        _state.value = _state.value?.copy(isLocationAvailable = treeCapturer.pinLocation())
```

Comment:
Please only update the state with a single copy method.

Reply:
Not sure I get this. Do you mean each method should only update a single state object?

Reply:
Each time the state object is updated, it refreshes the screen. In this case you update it once per field, which updates the screen two times here.

To avoid this you can update the _state.value once with a single copy method that updates all the required fields in it.

Reply:
Oh I get this now.